src/test/java/BasicMAPF/Solvers/CBS/CBS_SolverTest.java

```diff
@@ -416,4 +418,27 @@ void sharedGoals(){
             assertNull(solution);
         }
     }
+
+    @Test
+    void worksWithTMAPFPaths() {
+        I_Solver CBSt = new CBS_Solver(null, null, null, null, null, null, null, null, TransientMAPFBehaviour.transientMAPF);
+        Agent agentXMoving = new Agent(0, coor42, coor02, 1);
+        Agent agentYMoving = new Agent(1, coor10, coor12, 1);
+        MAPF_Instance testInstance = new MAPF_Instance("testInstance", mapEmpty, new Agent[]{agentXMoving, agentYMoving});
+
+        Solution solvedNormal = cbsSolver.solve(testInstance, new RunParametersBuilder().setTimeout(1000L).setInstanceReport(instanceReport).createRP());
+        assertTrue(solvedNormal.solves(testInstance));
+        assertEquals(4 + 4, solvedNormal.sumIndividualCosts());
+        assertEquals(6, solvedNormal.makespan());
+
+        Solution solvedCBSt = CBSt.solve(testInstance, new RunParametersBuilder().setTimeout(1000L).setInstanceReport(instanceReport).createRP());
+        assertTrue(solvedCBSt.solves(testInstance));
+        assertEquals(4 + 3, solvedCBSt.sumIndividualCosts()); // normal SOC function
+        assertEquals(4 + 2, solvedCBSt.sumServiceTimes()); // TMAPF cost function
+        assertEquals(4, solvedCBSt.makespan()); // makespan (normal)
+        assertEquals(4, solvedCBSt.makespanServiceTime()); // makespan (TMAPF)
+
+        System.out.println(solvedNormal);
+        System.out.println(solvedCBSt);
+    }
```

Comment:
Can you also add the example from the paper?

Reply:
Use a `MapFactory.newArbitraryGraphMap` for it because it isn't on a grid

Reply:
Done :)